src/qml/qgismobileapp.qml

```diff
@@ -2246,8 +2282,11 @@ ApplicationWindow {
               mapCanvasMap.unfreeze('follow');
               gnssButton.followActive = false;
               gnssButton.followOrientationActive = false;
-              displayToast(qsTr("Canvas stopped following location"));
+              mapSettingsConnections.autoReFollowLocation = true;
+              displayToast(qsTr('Follow location again'), 'info', qsTr('Unlock'), mapSettingsConnections.cancelAutoFollowLocation, mapSettingsConnections.followLocation, true);
             }
+          } else if (mapSettingsConnections.autoReFollowLocation) {
+            displayToast(qsTr('Follow location again'), 'info', qsTr('Unlock'), mapSettingsConnections.cancelAutoFollowLocation, mapSettingsConnections.followLocation, true);
```

Comment:
OK, this is where it should be called.

If we still need a autoReFollowLocation, let's add it as a property to the gnssButton (where other related properties live), and let's rename this to autoRefollow, smaller and clear enough. No **ReF**ollow :) )